src/Redmine/Api/Attachment.php

```diff
@@ -41,13 +41,15 @@ public function download($id)
      * Upload a file to redmine.
      *
      * @see http://www.redmine.org/projects/redmine/wiki/Rest_api#Attaching-files
-     *
+     * available $params :
+	 * filename: filename of the attachment
+	 * 
      * @param string $attachment the attachment content
-     *
+     * @param array $params optional parameters to be passed to the api 
      * @return array information about the attachment
      */
-    public function upload($attachment)
+    public function upload($attachment, $params = [])
     {
-        return $this->post('/uploads.json', $attachment);
+        return $this->post('/uploads.json?'.http_build_query($params), $attachment);
```

Comment:
this change makes the unit tests to fail, could you fix that please?
https://travis-ci.org/kbsali/php-redmine-api/jobs/517319254

Reply:
Hi,
Sorry, I should have check unit tests before requesting merge. I fixed that.

Reply:
Awesome, thanks @Ridiane !